### PR TITLE
Fix/overlay growing margin, issue 2323 

### PR DESCRIPTION
**Pull request by ka9de**
## What I did

1. fix the _keepBodySize to return instantly if controller uses local placement
2. wrote a test to confirm this

### CODE REVIEW

File: .changeset/avoid-growing-margin.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@lion/ui': patch
+---
+
+fix(2323): avoid growing margin when using overlays with tooltips
```
Comment (tlouisse) [inline .changeset/avoid-growing-margin.md:5]:
```suggestion
fix: [overlays] avoid growing margin when using overlays with tooltips
```

File: packages/ui/components/overlays/src/OverlayController.js [hunk 2]
```diff
@@ -787,6 +787,10 @@ export class OverlayController extends EventTarget {
    * @protected
    */
   _keepBodySize({ phase }) {
+    if (this.placementMode !== 'global') {
```
Comment (tlouisse) [inline packages/ui/components/overlays/src/OverlayController.js:790]:
```suggestion
    if (!this.preventsScroll) {
```

File: packages/ui/components/overlays/test/OverlayController.test.js [hunk 3]
```diff
@@ -1764,4 +1764,20 @@ describe('OverlayController', () => {
       );
     });
   });
+
+  describe('run _keepBodySize only with global placement', () => {
+    /**
+     * @type {OverlayController}
+     */
+    const overlayControllerLocal = new OverlayController({
+      ...withLocalTestConfig(),
```
Comment (tlouisse) [inline packages/ui/components/overlays/test/OverlayController.test.js:1773]:
```suggestion
      ...withLocalTestConfig(),
      preventsScroll: true,
```

File: packages/ui/components/overlays/test/OverlayController.test.js [hunk 4]
```diff
@@ -1764,4 +1764,20 @@ describe('OverlayController', () => {
...
+    });
+
+    it('should not run with local placement', async () => {
+      await overlayControllerLocal.show();
+
+      expect(overlayControllerLocal.__bodyMarginRightInline).to.equal(undefined);
+      expect(overlayControllerLocal.__bodyMarginRight).to.equal(undefined);
+    });
```
Comment (tlouisse) [inline packages/ui/components/overlays/test/OverlayController.test.js:1781]: Looks great, maybe you can also test here for the `preventsScroll: true` case...

```
      overlayControllerLocal.updateConfig({ preventsScroll: true }); 

      expect(overlayControllerLocal.__bodyMarginRightInline).to.not.equal(undefined);
      expect(overlayControllerLocal.__bodyMarginRight).to.not.equal(undefined);

```

**Review by tlouisse — CHANGES_REQUESTED** [review]
Thanks for the PR!

The changelog was correct indeed 👍 

Can you look at the suggestions? It would also be nice if you could squash all commits into one (but we can also do it while merging...)

When these latest changes are implemented, we will merge it 💪

**Comment by tlouisse** [comment]
closes https://github.com/ing-bank/lion/issues/2323

File: packages/ui/components/overlays/test/OverlayController.test.js [hunk 5]
```diff
@@ -1764,4 +1764,20 @@ describe('OverlayController', () => {
...
+     */
+    const overlayControllerLocal = new OverlayController({
+      ...withLocalTestConfig(),
+    });
+
+    it('should not run with local placement', async () => {
+      await overlayControllerLocal.show();
+
+      expect(overlayControllerLocal.__bodyMarginRightInline).to.equal(undefined);
+      expect(overlayControllerLocal.__bodyMarginRight).to.equal(undefined);
```
Comment (tlouisse) [inline packages/ui/components/overlays/test/OverlayController.test.js:1780]:
```suggestion
      expect(overlayControllerLocal.__bodyMarginRight).to.equal(undefined);
      
      overlayControllerLocal.updateConfig({ preventsScroll: true }); 
      expect(overlayControllerLocal.__bodyMarginRightInline).to.not.equal(undefined);
      expect(overlayControllerLocal.__bodyMarginRight).to.not.equal(undefined);
```

**Review by ka9de — COMMENTED** [review]
I fixed some linting problem and updated the tests with some text

**Review by tlouisse — APPROVED** [review]

**Comment by tlouisse** [comment]
Thanks for the PR!

**Pull request merged**